### PR TITLE
Fixed #2626 fixed leaflet measure tool

### LINKED ISSUE #2626: length measurement in nautical miles (along great circles)
### Description
see [this issue](https://github.com/geosolutions-it/austrocontrol-C125/issues/4)

### Other useful information (optional):
for this issue non need to add the add as annotation button

### PR DESCRIPTION
**Pull request by MV88**
## Description
This issue aligns leaflet measure tool to the openlayers one with one exception:
The tooltip in leaflet appears only while drawing and it is not attached to the map when the draw process ends.
Also the layout is different (css related)
for this will be opened a new issue in the future

## Issues
Fix #2626 comments

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)

 - [ ] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [x] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
leaflet measure tool behaviour is different from openlayers one

**What is the new behavior?**
it is similar with the tooltip ecxeption.

**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [x] Yes
 - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...
**note:**
I removed some functions (mtoft, mtomi, etc) in order to enhance modularity.
if some of this function are used in some project we need to changed them with new function convertUom or add new uom conversion coefficient

**Other information**:

This PR contains some overrides of leaflet draw handlers and L.GeometryUtil
- [_getMeasurementString](https://github.com/Leaflet/Leaflet.draw/blob/c6af99b761c5d9900bf5a3e2e1d8827bbde29542/src/draw/handler/Draw.Polygon.js#L81)

- [_getMeasurementString](https://github.com/Leaflet/Leaflet.draw/blob/c6af99b761c5d9900bf5a3e2e1d8827bbde29542/src/draw/handler/Draw.Polyline.js#L515)

- [readableDistance](https://github.com/Leaflet/Leaflet.draw/blob/c6af99b761c5d9900bf5a3e2e1d8827bbde29542/src/ext/GeometryUtil.js#L109)

- [readableArea](https://github.com/Leaflet/Leaflet.draw/blob/c6af99b761c5d9900bf5a3e2e1d8827bbde29542/src/ext/GeometryUtil.js#L66)

### CODE REVIEW

**Comment by MV88** [comment]
I was not able to test the overridden function of leaflet.draw. 
maybe you can suggest me how @mbarto @offtherailz

File: web/client/components/map/leaflet/MeasurementSupport.jsx [hunk 1]
```diff
@@ -2,21 +2,217 @@ const PropTypes = require('prop-types');
...
+    } else {
+        units = 'yards';
+    }
+    if (options.geomType === 'Bearing') {
+        return options.bearing;
+    }
+    switch (units) {
+        // show meters when distance is < 1km, then show km
+        case 'metric':
+            if (options.distinctMeasure) {
```
Comment (mbarto) [inline web/client/components/map/leaflet/MeasurementSupport.jsx]: Why distinctMeasure? Explain the meaning please

File: web/client/components/map/leaflet/MeasurementSupport.jsx [hunk 2]
```diff
@@ -2,21 +2,217 @@ const PropTypes = require('prop-types');
...
+    slice
+} = require('lodash');
+const {
+    reproject,
+    calculateAzimuth,
+    calculateDistance,
+    transformLineToArcs
+} = require('../../../utils/CoordinatesUtils');
+const {
+    sqmTosqft,
```
Comment (mbarto) [inline web/client/components/map/leaflet/MeasurementSupport.jsx]: Can't we translate all these function into one (e.g. convertUoM(value, source, dest) ?

File: web/client/components/map/leaflet/MeasurementSupport.jsx [hunk 3]
```diff
@@ -2,21 +2,217 @@ const PropTypes = require('prop-types');
...
+            if (a >= 3097600) { // 3097600 square yards in 1 square mile
+                areaStr = L.GeometryUtil.formattedNumber(a / 3097600, p.mi) + ' mi²';
+            } else if (a >= 4840) { // 4840 square yards in 1 acre
+                areaStr = L.GeometryUtil.formattedNumber(a / 4840, p.ac) + ' acres';
+            } else {
+                areaStr = L.GeometryUtil.formattedNumber(a, p.yd) + ' yd²';
+            }
+        }
+    } else {
+        if (unit === "sqkm") {
```
Comment (mbarto) [inline web/client/components/map/leaflet/MeasurementSupport.jsx]: Can we translate all these if(s) into a functional approach?

File: web/client/components/map/leaflet/MeasurementSupport.jsx (hunk 1 above)
Comment (MV88) [reply web/client/components/map/leaflet/MeasurementSupport.jsx]: i meant when you use a treshold to switch from meters and kilometers for example (like leaflet).
i'll change it to useTreshold

Comment (mbarto) [reply web/client/components/map/leaflet/MeasurementSupport.jsx]: ok, that's more clear

File: web/client/components/map/leaflet/MeasurementSupport.jsx (hunk 3 above)
Comment (MV88) [reply web/client/components/map/leaflet/MeasurementSupport.jsx]: indeed

File: web/client/components/map/leaflet/MeasurementSupport.jsx (hunk 2 above)
Comment (MV88) [reply web/client/components/map/leaflet/MeasurementSupport.jsx]: there are already getFormattedLength and getFormattedArea which takes source by default in meters and square meters

Comment (mbarto) [reply web/client/components/map/leaflet/MeasurementSupport.jsx]: those are for formatting, I mean a function for converting between uoms

**Pull request merged**